flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/optimize/program/FlinkBatchProgram.scala

```diff
@@ -73,6 +74,8 @@ object FlinkBatchProgram {
           .build(), "convert table references after sub-queries removed")
         .build()
     )
+    // convert time indicators
+    chainedProgram.addLast(TIME_INDICATOR, new FlinkRelTimeIndicatorProgram)
```

Comment:
We put TIME_INDICATOR phase under decorrelate in streaming programs, I think we should keep the same order in batch program to put the phase after decorrelate.

Reply:
Good catch!